zig setCold removed on nightly

related to https://github.com/ziglang/zig/issues/21148
log:
```
install
└─ install arocc
   └─ zig build-exe arocc ReleaseFast native 9 errors
src/aro/Parser.zig:393:5: error: invalid builtin function: '@setCold'
    @setCold(true);
    ^~~~~~~~~~~~~~
src/aro/Parser.zig:398:5: error: invalid builtin function: '@setCold'
    @setCold(true);
    ^~~~~~~~~~~~~~
src/aro/Parser.zig:415:5: error: invalid builtin function: '@setCold'
    @setCold(true);
    ^~~~~~~~~~~~~~
src/aro/Parser.zig:420:5: error: invalid builtin function: '@setCold'
    @setCold(true);
    ^~~~~~~~~~~~~~
src/aro/Parser.zig:654:5: error: invalid builtin function: '@setCold'
    @setCold(true);
    ^~~~~~~~~~~~~~
src/aro/Driver/Filesystem.zig:7:5: error: invalid builtin function: '@setCold'
    @setCold(true);
    ^~~~~~~~~~~~~~
src/aro/Driver/Filesystem.zig:19:5: error: invalid builtin function: '@setCold'
    @setCold(true);
    ^~~~~~~~~~~~~~
src/aro/Driver/Filesystem.zig:38:5: error: invalid builtin function: '@setCold'
    @setCold(true);
    ^~~~~~~~~~~~~~
src/aro/Driver/Filesystem.zig:48:5: error: invalid builtin function: '@setCold'
    @setCold(true);
    ^~~~~~~~~~~~~~
error: the following command failed with 9 compilation errors:
/usr/lib/zig/9999/bin/zig build-exe -fsingle-threaded -OReleaseFast --dep aro -Mroot=/home/misile/arocc/src/main.zig --dep system_defaults --dep build_options --dep backend --dep Builtins/Builtin.def --dep Attribute/names.def --dep Diagnostics/messages.def -Maro=/home/misile/arocc/src/aro.zig -Msystem_defaults=/home/misile/arocc/.zig-cache/c/1b18a22cd7666b4de994d135ecafc9a9/options.zig -Mbuild_options=/home/misile/arocc/.zig-cache/c/3a9636679c789b6890c6829c1e31eda4/options.zig --dep zig --dep build_options -Mbackend=/home/misile/arocc/src/backend.zig -MBuiltins/Builtin.def=/home/misile/arocc/.zig-cache/o/312142940bdd855eed805ccc9527c189/Builtin.zig -MAttribute/names.def=/home/misile/arocc/.zig-cache/o/0efd2a72c2a79037f32c2204d0c600ec/names.zig -MDiagnostics/messages.def=/home/misile/arocc/.zig-cache/o/1526c758785a0eeed33e38d9ca57c7aa/messages.zig -Mzig=/home/misile/arocc/deps/zig/lib.zig -lc --cache-dir /home/misile/arocc/.zig-cache --global-cache-dir /home/misile/.cache/zig --name arocc --zig-lib-dir /usr/lib/zig/9999/lib/ --listen=-
Build Summary: 6/9 steps succeeded; 1 failed
install transitive failure
└─ install arocc transitive failure
   └─ zig build-exe arocc ReleaseFast native 9 errors
error: the following build command failed with exit code 1:
/home/misile/arocc/.zig-cache/o/3b427bcce4387cba3c4e421da7716ec2/build /usr/lib/zig/9999/bin/zig /usr/lib/zig/9999/lib /home/misile/arocc /home/misile/arocc/.zig-cache /home/misile/.cache/zig --seed 0x96ddc892 -Z7e71a533c5e525b4 -Doptimize=ReleaseFast
```
version: `0.14.0-dev.1342+1a178d499`

--- a/deps/zig/arch/x86_64/Lower.zig
+++ b/deps/zig/arch/x86_64/Lower.zig
@@ -271,7 +271,7 @@ pub fn lowerMir(lower: *Lower, index: Mir.Inst.Index) Error!struct {
 }
 
 pub fn fail(lower: *Lower, comptime format: []const u8, args: anytype) Error {
-    @setCold(true);
+    @branchHint(.cold);
     assert(lower.err_msg == null);
     lower.err_msg = try std.fmt.allocPrint(lower.gpa, format, args);
     return error.LowerFail;

--- a/src/aro/Driver/Filesystem.zig
+++ b/src/aro/Driver/Filesystem.zig
@@ -4,7 +4,7 @@ const builtin = @import("builtin");
 const is_windows = builtin.os.tag == .windows;
 
 fn readFileFake(entries: []const Filesystem.Entry, path: []const u8, buf: []u8) ?[]const u8 {
-    @setCold(true);
+    @branchHint(.cold);
     for (entries) |entry| {
         if (mem.eql(u8, entry.path, path)) {
             const len = @min(entry.contents.len, buf.len);
@@ -16,7 +16,7 @@ fn readFileFake(entries: []const Filesystem.Entry, path: []const u8, buf: []u8) 
 }
 
 fn findProgramByNameFake(entries: []const Filesystem.Entry, name: []const u8, path: ?[]const u8, buf: []u8) ?[]const u8 {
-    @setCold(true);
+    @branchHint(.cold);
     if (mem.indexOfScalar(u8, name, '/') != null) {
         @memcpy(buf[0..name.len], name);
         return buf[0..name.len];
@@ -35,7 +35,7 @@ fn findProgramByNameFake(entries: []const Filesystem.Entry, name: []const u8, pa
 }
 
 fn canExecuteFake(entries: []const Filesystem.Entry, path: []const u8) bool {
-    @setCold(true);
+    @branchHint(.cold);
     for (entries) |entry| {
         if (mem.eql(u8, entry.path, path)) {
             return entry.executable;
@@ -45,7 +45,7 @@ fn canExecuteFake(entries: []const Filesystem.Entry, path: []const u8) bool {
 }
 
 fn existsFake(entries: []const Filesystem.Entry, path: []const u8) bool {
-    @setCold(true);
+    @branchHint(.cold);
     var buf: [std.fs.max_path_bytes]u8 = undefined;
     var fib = std.heap.FixedBufferAllocator.init(&buf);
     const resolved = std.fs.path.resolvePosix(fib.allocator(), &.{path}) catch return false;

--- a/src/aro/Parser.zig
+++ b/src/aro/Parser.zig
@@ -390,12 +390,12 @@ fn errExpectedToken(p: *Parser, expected: Token.Id, actual: Token.Id) Error {
 }
 
 pub fn errStr(p: *Parser, tag: Diagnostics.Tag, tok_i: TokenIndex, str: []const u8) Compilation.Error!void {
-    @setCold(true);
+    @branchHint(.cold);
     return p.errExtra(tag, tok_i, .{ .str = str });
 }
 
 pub fn errExtra(p: *Parser, tag: Diagnostics.Tag, tok_i: TokenIndex, extra: Diagnostics.Message.Extra) Compilation.Error!void {
-    @setCold(true);
+    @branchHint(.cold);
     const tok = p.pp.tokens.get(tok_i);
     var loc = tok.loc;
     if (tok_i != 0 and tok.id == .eof) {
@@ -412,12 +412,12 @@ pub fn errExtra(p: *Parser, tag: Diagnostics.Tag, tok_i: TokenIndex, extra: Diag
 }
 
 pub fn errTok(p: *Parser, tag: Diagnostics.Tag, tok_i: TokenIndex) Compilation.Error!void {
-    @setCold(true);
+    @branchHint(.cold);
     return p.errExtra(tag, tok_i, .{ .none = {} });
 }
 
 pub fn err(p: *Parser, tag: Diagnostics.Tag) Compilation.Error!void {
-    @setCold(true);
+    @branchHint(.cold);
     return p.errExtra(tag, p.tok_i, .{ .none = {} });
 }
 
@@ -651,7 +651,7 @@ fn pragma(p: *Parser) Compilation.Error!bool {
 
 /// Issue errors for top-level definitions whose type was never completed.
 fn diagnoseIncompleteDefinitions(p: *Parser) !void {
-    @setCold(true);
+    @branchHint(.cold);
 
     const node_slices = p.nodes.slice();
     const tags = node_slices.items(.tag);

